File: JAVA/java/src/SameTree.java
class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;

    TreeNode(int x) {
        val = x;
    }
}

public class SameTree {

    public boolean isSameTree(TreeNode p, TreeNode q) {
        if (p == null && q == null) {
            return true;
        }
        if (p == null || q == null) {
            return false;
        }
        return p.val == q.val && isSameTree(p.left, q.left) && isSameTree(p.right, q.right);
    }

    public static void main(String[] args) {
        // Create TreeNode objects from the input lists
        TreeNode p = new TreeNode(1);
        p.left = new TreeNode(2);

        TreeNode q = new TreeNode(1);
        q.right = new TreeNode(2);

        // Create an instance of SameTree
        SameTree solution = new SameTree();

        // Call the method with the created TreeNode objects
        boolean result = solution.isSameTree(p, q);

        // Print the result
        System.out.println(result);
    }
}

File: python/Same Tree.py
from typing import Optional

class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

def isSameTree(p: Optional[TreeNode], q: Optional[TreeNode]) -> bool:
    if not p and not q:
        return True
    if not p or not q:
        return False
    if p.val != q.val:
        return False
    else:
        return p.val == q.val and isSameTree(p.left, q.left) and isSameTree(p.right, q.right)

# Create TreeNode objects from the input lists
p = TreeNode(1, TreeNode(2))
q = TreeNode(1, None, TreeNode(2))

# Call the function with the created TreeNode objects
result = isSameTree(p, q)

# Print the result
print(result)
